src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -898,7 +918,12 @@ public struct OperationAnalysisContext
         /// <see cref="IOperation"/> that is the subject of the analysis.
         /// </summary>
         public IOperation Operation { get { return _operation; } }
-        
+
+        /// <summary>
+        /// <see cref="CodeAnalysis.SemanticModel"/> that can provide semantic information about the <see cref="IOperation.Syntax"/> associated with the <see cref="Operation"/>.
+        /// </summary>
+        public SemanticModel SemanticModel { get { return _semanticModel; } }
```

Comment:
In theory the semantic model is not useful to an operation action. The main reason for making the syntax node available is to be able to get location information. Now that we're thinking about mapping IL to IOperation, replacing the syntax node property from IOperation with a location property might be reasonable. If we do that, then a semantic model would be completely useless to an author.

So, I don't think the semantic model property should be public here.